pkg/nuclide-swift/lib/taskrunner/SwiftPMTaskRunner.js

```diff
@@ -0,0 +1,185 @@
+'use babel';
+/* @flow */
+
+/*
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ */
+
+import type {Task, TaskMetadata} from '../../../nuclide-task-runner/lib/types';
+import type {Level, Message} from '../../../nuclide-console/lib/types';
+import type {SwiftPMTaskRunnerStoreState} from './SwiftPMTaskRunnerStoreState';
+
+import invariant from 'assert';
+import {Observable, Subject} from 'rxjs';
+import {Dispatcher} from 'flux';
+import {CompositeDisposable} from 'atom';
+import {React} from 'react-for-atom';
+import fsPromise from '../../../commons-node/fsPromise';
+import {DisposableSubscription} from '../../../commons-node/stream';
+import {observeProcess, safeSpawn} from '../../../commons-node/process';
+import {observableToTaskInfo} from '../../../commons-node/observableToTaskInfo';
+import SwiftPMTaskRunnerStore from './SwiftPMTaskRunnerStore';
+import SwiftPMTaskRunnerActions from './SwiftPMTaskRunnerActions';
+import {buildCommand, testCommand} from './SwiftPMTaskRunnerCommands';
+import {
+  SwiftPMTaskRunnerBuildTaskMetadata,
+  SwiftPMTaskRunnerTestTaskMetadata,
+  SwiftPMTaskRunnerTaskMetadata,
+} from './SwiftPMTaskRunnerTaskMetadata';
+import SwiftPMTaskRunnerToolbar from './toolbar/SwiftPMTaskRunnerToolbar';
+import {SwiftIcon} from '../ui/SwiftIcon';
+
+/**
+ * The primary controller for spawning SwiftPM tasks, such as building a
+ * package, or running its tests.
+ *
+ * This class conforms to Nuclide's TaskRunner interface, and makes use of the
+ * Flux design pattern. This class is responsible for kicking off SwiftPM tasks
+ * such as building a package. How it builds the package is determined by the
+ * state of the SwiftPMTaskRunnerToolbar -- the path to the package, whether a
+ * build path is specified, etc. -- and that state is maintained by the
+ * SwiftPMTaskRunnerStore. Updates to the toolbar UI options trigger actions,
+ * defined in SwiftPMTaskRunnerActions, which update the state of the store.
+ * Actions are routed to the store via a Flux.Dispatcher (instantiated by this
+ * class).
+ */
+export class SwiftPMTaskRunner {
+  id: string;
+  name: string;
+  _disposables: CompositeDisposable;
+  _store: SwiftPMTaskRunnerStore;
+  _actions: SwiftPMTaskRunnerActions;
+  _taskList: Observable<Array<TaskMetadata>>;
+  _outputMessages: Subject<Message>;
+
+  constructor(initialState: ?SwiftPMTaskRunnerStoreState) {
+    this.id = 'swiftpm';
+    this.name = 'Swift';
+
+    const dispatcher = new Dispatcher();
+    this._store = new SwiftPMTaskRunnerStore(dispatcher, initialState);
+    this._actions = new SwiftPMTaskRunnerActions(dispatcher);
```

Comment:
Do you think you could lazily create these when they're used instead? Since we're using inline imports, that would save us having to do the IO for loading the modules until/unless the user actually selected a Swift task.